File: app/src/main/java/ru/vpiska/rating/RatingAdapter.java
package ru.vpiska.rating;


import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;

import ru.vpiska.R;
import ru.vpiska.party.PartyActivity;
import ru.vpiska.profile.OtherProfileActivity;

public class RatingAdapter extends ArrayAdapter<Rating> {
    private LayoutInflater inflater;
    private int layout;
    private ArrayList<Rating> ratingList;
    private Context mContext = getContext();

    RatingAdapter(Context context, int resource, ArrayList<Rating> rating) {
        super(context, resource, rating);
        this.ratingList = rating;
        this.layout = resource;
        this.inflater = LayoutInflater.from(context);
    }

    @SuppressLint("SetTextI18n")
    @NonNull
    public View getView(int position, View convertView,@NonNull ViewGroup parent) {

        final ViewHolder viewHolder;
        if (convertView == null) {
            convertView = inflater.inflate(this.layout, parent, false);
            viewHolder = new ViewHolder(convertView);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }
        final Rating product = ratingList.get(position);

        viewHolder.nameView.setText(product.getName());
        viewHolder.ratingView.setText("Рейтинг: " + product.getRating());


        viewHolder.linkButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(mContext, OtherProfileActivity.class);
                intent.putExtra("id_user", product.getId());
                //Toast.makeText(getContext(),  product.getId(), Toast.LENGTH_LONG).show();
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_MULTIPLE_TASK);
                mContext.startActivity(intent);



            }
        });


        return convertView;
    }



    private class ViewHolder {
        final Button linkButton;
        final TextView nameView,ratingView;

        ViewHolder(View view) {
            linkButton = (Button) view.findViewById(R.id.linkButton);
            nameView = (TextView) view.findViewById(R.id.nameView);
            ratingView = (TextView) view.findViewById(R.id.ratingView);

        }
    }
}

File: app/src/main/java/ru/vpiska/rating/RatingAdapter.kt
package ru.vpiska.rating


import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.TextView
import ru.vpiska.R
import ru.vpiska.profile.OtherProfileActivity
import java.util.*

class RatingAdapter internal constructor(context: Context, private val layout: Int, private val ratingList: ArrayList<Rating>) : ArrayAdapter<Rating>(context, layout, ratingList) {
    private val inflater: LayoutInflater = LayoutInflater.from(context)
    private val mContext = getContext()

    @SuppressLint("SetTextI18n")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView

        val viewHolder: ViewHolder
        if (convertView == null) {
            convertView = inflater.inflate(this.layout, parent, false)
            viewHolder = ViewHolder(convertView!!)
            convertView.tag = viewHolder
        } else {
            viewHolder = convertView.tag as ViewHolder
        }
        val product = ratingList[position]

        viewHolder.nameView.text = product.name
        viewHolder.ratingView.text = "Рейтинг: " + product.rating


        viewHolder.linkButton.setOnClickListener {
            val intent = Intent(mContext, OtherProfileActivity::class.java)
            intent.putExtra("id_user", product.id)
            //Toast.makeText(getContext(),  product.getId(), Toast.LENGTH_LONG).show();
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_MULTIPLE_TASK
            mContext.startActivity(intent)
        }


        return convertView
    }


    private inner class ViewHolder internal constructor(view: View) {
        internal val linkButton: Button = view.findViewById(R.id.linkButton)
        internal val nameView: TextView = view.findViewById(R.id.nameView)
        internal val ratingView: TextView = view.findViewById(R.id.ratingView)

    }
}